### PR TITLE
Game creation rework

### PR DESCRIPTION
**Pull request by NicholasBatesNZ**
Refactored some stuff to make a potential multiplayer effort easier, most notably `WorldConfig` usages.
Also puts every mode through the loading screen, not just continue game. Improves the android ux as it previously looked frozen when loading a new game.

# Testing
Play the game. Tutorial, new game and continue game should work as normal.

### CODE REVIEW

File: engine/src/main/java/org/destinationsol/game/SaveManager.java [hunk 1]
```diff
@@ -46,10 +46,11 @@
 import java.io.UnsupportedEncodingException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 public class SaveManager {
 
-    private static Logger logger = LoggerFactory.getLogger(SaveManager.class);
+    private static final Logger logger = LoggerFactory.getLogger(SaveManager.class);
```
Comment (Adrijaned) [inline engine/src/main/java/org/destinationsol/game/SaveManager.java:53]: AllCaps? Or are we not running that convention? Been a while...

File: engine/src/main/java/org/destinationsol/game/SolGame.java [hunk 2]
```diff
@@ -107,6 +107,9 @@
 
     public SolGame(String shipName, boolean isTutorial, boolean isNewGame, CommonDrawer commonDrawer, Context context,
                    WorldConfig worldConfig) {
+        // TODO: make this non-static
+        FactionInfo staticIgnore = new FactionInfo();
```
Comment (Adrijaned) [inline engine/src/main/java/org/destinationsol/game/SolGame.java:111]: Couldn't you, like, at least make a static `init()` method to call instead? This is ugly.

File: engine/src/main/java/org/destinationsol/game/SaveManager.java (hunk 1 above)
Comment (NicholasBatesNZ) [reply engine/src/main/java/org/destinationsol/game/SaveManager.java:53]: Good question... I think we are in some places but not others...

**Comment by NicholasBatesNZ** [comment]
#514

**Pull request closed without merging**